metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1108,19 +1116,23 @@ private Errors validateAlterPartitionData(
 
             return INVALID_UPDATE_VERSION;
         }
-        int[] newIsr = Replicas.toArray(partitionData.newIsr());
+
+        int[] newIsr = partitionData.newIsrWithEpochs().stream()
+            .map(brokerState -> brokerState.brokerId()).collect(Collectors.toList())
+            .stream().mapToInt(Integer::intValue).toArray();
```

Comment:
Do we really need the intermediate `.collect(Collectors.toList())`? If it is, it may be better to add another method to `Replicas`: `Replicas.toArray(partitionData.newIsrWithEpochs())`. There we could initialize an array with the right size and set the values.

Reply:
You are right, it is not needed. Thanks for pointing this out.